test/repo_test.rb

```diff
@@ -804,6 +802,11 @@ def verify_subtrees
   end
 
   def test_checkout_tree_with_commit
+    # the test repo has an unclean status. apparently libgit2 on *nix does not
```

Comment:
Another interesting thing I don't know how to resolve. The libgit2 test repo has modifications initially, confirmedon linux and windows. Apparently on *nix checking out another pointer isn't an issue here, on Windows it is for some reason.